presto-main/src/main/java/io/prestosql/sql/planner/plan/TableScanNode.java

```diff
@@ -42,48 +40,43 @@
     private final List<Symbol> outputSymbols;
     private final Map<Symbol, ColumnHandle> assignments; // symbol -> column
 
-    private final Optional<TableLayoutHandle> tableLayout;
-
     // Used during predicate refinement over multiple passes of predicate pushdown
     // TODO: think about how to get rid of this in new planner
     private final TupleDomain<ColumnHandle> currentConstraint;
 
     private final TupleDomain<ColumnHandle> enforcedConstraint;
 
+    public static TableScanNode newInstance(
```

Comment:
This is a strange design, which I think predates this change.  Maybe add a comment on why this factory is different from the constructor with the same signature.

Reply:
Ah, thanks for pointing it out. I need to see if I can get rid of it. Without it, two of the constructors had the same shape (but ended up doing different things). I may be able to consolidate them now that we don’t have table layouts.

Reply:
It turns out I can't. I tried making the static factory method the one that's used to deserialized from json, but it needs to set some fields to null, which is disallowed by the full constructor. Oh well.. I added a comment to clarify.

Reply:
But we used to have such constructor and it used to work. What was wrong with it?